app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+
+        Some attributes in the model are different from the one for the serializer, this is the
+        mapping between the two.
+
+        Args:
+            key: string
+                key attribute to map into json key
+
+        Returns:
+            string, json key mapped
+        '''
+        return self.key_mapping.get(key, key)
+
+    def create(self):
+        '''Create the sample in DB
+
+        Returns:
+            the DB sample (model object)
+        '''
+        self.model_instance = self.model_class(**self.attributes)
+        self.model_instance.full_clean()
+        self.model_instance.save()
+        return self.model_instance
+
+    @property
+    def json(self):
+        '''Returns a json serializable representation of the sample data
+
+        This json payload can then be used in the write API payload or to check
+        the read API payload.
+
+        Returns
+            A dictionary with the sample data
+        '''
+        return {
+            self.key_mapped(key): value for key,
+            value in self.get_attributes(remove_relations=False).items()
+        }
+
+    @property
+    def model(self):
+        '''Returns a django DB model object of sample data
+
+        If the data has not yet been created in DB, then it is created.
+
+        Returns:
+            model instance
+        '''
+        if not self.model_instance:
+            self.create()
+        return self.model_instance
+
+
+class LinkSample(SampleData):
+    '''Link Sample Data base class
+    '''
+    key_mapping = {'link_type': 'type'}
+    optional_fields = ['title', 'link_type']
+
+    def __init__(self, sample='link-1', rel=None, **kwargs):
+        '''Create a link sample data
+        '''
+        super().__init__(sample=sample, rel=rel, **kwargs)
+
+
+class ProviderSample(SampleData):
+    '''Collection's Provider Sample Data
+    '''
+    model_class = Provider
+    sample_name = 'provider'
+    samples_dict = provider_samples
+    optional_fields = ['description', 'roles', 'url']
+
+    def __init__(self, sample='provider-1', name=None, **kwargs):
+        '''Create a collection's provider sample data
+        '''
+        super().__init__(sample=sample, name=name, **kwargs)
+
+
+class CollectionLinkSample(LinkSample):
+    '''Collection's Link Sample Data
+    '''
+    model_class = CollectionLink
+    sample_name = 'collection-link'
+    samples_dict = collection_link_samples
+
+
+class CollectionSample(SampleData):
+    '''Collection Sample Data
+    '''
+    model_class = Collection
+    sample_name = 'collection'
+    samples_dict = collection_samples
+    key_mapping = {'name': 'id'}
+    optional_fields = ['title', 'providers', 'links']
+
+    def __init__(self, sample='collection-1', name=None, **kwargs):
+        '''Create a collection sample data
+
+        Args:
+            sample: string
+                Name of the sample based to use, see tests.sample_data.collection_samples
+            required_only: bool
```

Comment:
`required_only` is not an argument of `__init__`, probably docstring has not been updated after changes?

Reply:
It is but simply hidden in the `kwargs` and used by the base class.